ci/ci/github.py

```diff
@@ -385,18 +387,19 @@ async def _update_github(self, gh):
         await self._update_github_review_state(gh)
 
     @staticmethod
-    def _hail_github_status_from_statuses(statuses_json):
+    def _hail_github_status_from_statuses(statuses_json) -> Dict[str, GithubStatus]:
         statuses = statuses_json["statuses"]
-        hail_status = [s for s in statuses if s["context"] == GITHUB_STATUS_CONTEXT]
-        n_hail_status = len(hail_status)
-        if n_hail_status == 0:
-            return None
-        if n_hail_status == 1:
-            return hail_status[0]['state']
-        raise ValueError(
-            f'github sent multiple status summaries for our one '
-            f'context {GITHUB_STATUS_CONTEXT}: {hail_status}\n\n{statuses_json}'
-        )
+        hail_statuses = {}
+        for s in statuses:
+            context = s['context']
+            if context == GITHUB_STATUS_CONTEXT or context.startswith('hail-ci'):
+                if context in hail_statuses:
+                    raise ValueError(
+                        f'github sent multiple status summaries for our one '
+                        f'context {context}: {s}\n\n{statuses_json}'
+                    )
+                hail_statuses[context] = s['state']
+        return hail_statuses
```

Comment:
this can't possibly type check. Are we not running mypy on this file?

Reply:
I don't fully understand this either, but I ran mypy by hand and it does seem to be ok with this. I can add an assert to make this more clear?

Reply:
It seems the issue is that statuses_json is `Any` so all its children are `Any` and can be used for anything you like.

If you add a `:str` type annotation to context and status, mypy appropriately complains.

Reply:
Ah, I would be surprised that it's so permissive of Any, I would have assumed that would behave the same as str in this context.